core/lib/utils/application_manager.dart

```diff
@@ -0,0 +1,60 @@
+
+import 'package:core/utils/app_logger.dart';
+import 'package:core/utils/platform_info.dart';
+import 'package:device_info_plus/device_info_plus.dart';
+import 'package:fk_user_agent/fk_user_agent.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+
+class ApplicationManager {
+
+  final DeviceInfoPlugin _deviceInfoPlugin;
+
+  ApplicationManager(this._deviceInfoPlugin);
+
+  Future<PackageInfo> getPackageInfo() async {
+    final packageInfo = await PackageInfo.fromPlatform();
+    log('ApplicationManager::getPackageInto: $packageInfo');
+    return packageInfo;
+  }
+
+  Future<String> getVersion() async {
+    final version = (await getPackageInfo()).version;
+    log('ApplicationManager::getVersion: $version');
+    return version;
+  }
+
+  Future<String> getUserAgent() async {
```

Comment:
Please add unit test for `getUserAgent()`

Reply:
Done. Please see below commit